Test/E2ETests/E2ETests/Test452Base.cs

```diff
@@ -0,0 +1,1009 @@
+
+            // Deploy the docker cluster using Docker-Compose
+            //DockerUtils.ExecuteDockerComposeCommand("up -d --force-recreate --build", DockerComposeFileName);
+            DockerUtils.ExecuteDockerComposeCommand("up -d --build", DockerComposeFileName);
+            Thread.Sleep(5000);
+            DockerUtils.PrintDockerProcessStats("Docker-Compose -build");
+
+            // Populate dynamic properties of Deployed Apps like ip address.
+            PopulateIPAddresses();
+
+            bool webAppHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[WebAppName]);
+            bool webApiHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[WebApiName]);
+            bool ingestionHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[IngestionName]);
+
+            if (!(webAppHealthy && webApiHealthy && ingestionHealthy))
+            {
+                DockerUtils.ExecuteDockerComposeCommand("up -d --build", DockerComposeFileName);
+                Thread.Sleep(5000);
+                DockerUtils.PrintDockerProcessStats("Docker-Compose -build retry");
+
+                webAppHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[WebAppName]);
+                webApiHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[WebApiName]);
+                ingestionHealthy = HealthCheckAndRemoveImageIfNeeded(Apps[IngestionName]);
+            }            
+
+            Assert.IsTrue(webAppHealthy, "Web App is unhealthy");
+            Assert.IsTrue(webApiHealthy, "Web Api is unhealthy");
+            Assert.IsTrue(ingestionHealthy, "Ingestion is unhealthy");
+
+            dataendpointClient = new DataEndpointClient(new Uri("http://" + Apps[IngestionName].ipAddress));
+
+            Thread.Sleep(5000);
+            Trace.WriteLine(".Completed ClassInitialize:" + DateTime.UtcNow.ToLongTimeString());
+        }
+
+        private static void PopulateIPAddresses()
+        {
+            // Inspect Docker containers to get IP addresses
+            Apps[WebAppName].ipAddress = DockerUtils.FindIpDockerContainer(Apps[WebAppName].containerName);
+            Apps[WebApiName].ipAddress = DockerUtils.FindIpDockerContainer(Apps[WebApiName].containerName);
+            Apps[IngestionName].ipAddress = DockerUtils.FindIpDockerContainer(Apps[IngestionName].containerName);
+        }
+
+        private static void RestartAllTestAppContainers()
+        {
+            foreach (var app in Apps.Values)
+            {
+                DockerUtils.RestartDockerContainer(app.containerName);
+            }
+        }
+
+        public static void MyClassCleanupBase()
+        {
+            Trace.WriteLine("Started Class Cleanup:" + DateTime.UtcNow.ToLongTimeString());
+            //DockerUtils.ExecuteDockerComposeCommand("down", DockerComposeFileName);
+            //DockerUtils.RemoveDockerImage(Apps[WebAppName].imageName, true);
+            Trace.WriteLine("Completed Class Cleanup:" + DateTime.UtcNow.ToLongTimeString());
```

Comment:
should these lines be commented out?

Reply:
yes intentionally did that for speedy local runs. in the lab, a script is run to ensure full cleanups after every builds.